kernel/genlambda.ml

```diff
@@ -715,14 +758,22 @@ let rec lambda_of_constr cache env sigma c =
 and lambda_of_app cache env sigma f args =
   match kind f with
   | Const (kn, u as c) ->
-      let kn = get_alias env kn in
+      let kn, mask = get_alias env kn in
       let cb = lookup_constant kn env in
       begin match cb.const_body with
       | Primitive op -> lambda_of_prim env c op (lambda_of_args cache env sigma 0 args)
       | Def csubst -> (* TODO optimize if f is a proj and argument is known *)
         if cb.const_inline_code then lambda_of_app cache env sigma csubst args
         else
-          mkLapp (Lconst (kn, u)) (lambda_of_args cache env sigma 0 args)
+          (* Erase unused arguments *)
+          let mask = Array.of_list mask in
```

Comment:
I think you should just store `mask` as an array of boolean right from the start. Keeping it as a list and converting it again and again to an array is a waste of cycles and memory.

Reply:
Done.